DSP-759 Check if a Given Date Can Be Edited

tobiasschweizer opened the pull request:
resolves [DSP-759](https://dasch.myjetbrains.com/youtrack/issue/DSP-759)

Note that this PR renames `ValueTypeService` to `ValueService`.

tobiasschweizer commented:
@mdelez This PR handles precision and era, but not the calendar.

tobiasschweizer commented:
@mdelez sorry, too early to review: still missing the logic the prevents the editing button to show up

tobiasschweizer commented:
@mdelez Ok, now this PR is ready for review :-)

I decided to move the logic to value type service since it is also used there. It seemed weird to me to use logic coming from a component in a service.

tobiasschweizer commented on projects/dsp-ui/src/lib/viewer/services/value-type.service.ts:
```diff
@@ -78,6 +79,56 @@ export class ValueTypeService {
+    }
+
     /**
      * Equality checks with constants below are TEMPORARY until component is implemented.
```
@mdelez Maybe this comment should be adapted since there are also other reasons to make a value read-only. What do you think?

mdelez replied:
Maybe this service can also be renamed? It's becoming more general and is now more of a place to add helpful functions for the value components

mdelez reviewed (changes requested):
Nice! Just left a few comments to discuss

tobiasschweizer commented:
> Nice! Just left a few comments to discuss

I adapted the comment and renamed the service in https://github.com/dasch-swiss/dsp-ui-lib/pull/214/commits/b918f647d119dba0f5239dd1e7617bc1f77b932a.

I renamed the service to `ValueService` because it is not only about the value's type (the value instance is checked too).

Do you know if this service is used directly somewhere (outside the lib)? If yes, usage would break with the next release.

mdelez commented:
> Do you know if this service is used directly somewhere (outside the lib)? If yes, usage would break with the next release.

It is used [here](https://github.com/dasch-swiss/dsp-app/blob/4b2920d1200ea929131ba1e550f10233aba2dfb8/src/app/workspace/resource/resource-instance-form/select-properties/select-properties.component.ts#L36) in DSP-APP. However, the [PR](https://github.com/dasch-swiss/dsp-app/pull/324) that this code is in isn't merged yet so we can update it easily :)

tobiasschweizer commented:
> > Do you know if this service is used directly somewhere (outside the lib)? If yes, usage would break with the next release.
> 
> It is used [here](https://github.com/dasch-swiss/dsp-app/blob/4b2920d1200ea929131ba1e550f10233aba2dfb8/src/app/workspace/resource/resource-instance-form/select-properties/select-properties.component.ts#L36) in DSP-APP. However, the [PR](https://github.com/dasch-swiss/dsp-app/pull/324) that this code is in isn't merged yet so we can update it easily :)

Ok, great! I adapted the vars as well in https://github.com/dasch-swiss/dsp-ui-lib/pull/214/commits/338990600a5e8698425dcd2c42af5e4856646b1c

tobiasschweizer commented on projects/dsp-ui/src/lib/viewer/operations/display-edit/display-edit.component.spec.ts:
```diff
@@ -324,14 +322,11 @@ describe('DisplayEditComponent', () => {
             provide: MatDialogRef,
             useValue: {}
-        },
-        ValueService,
```
@mdelez I think there is no need to provide the service here since it is provided in root anyway. 
Importing `ReactiveFormsModule` fixes the injector problem with `FormBuilder`.

tobiasschweizer replied:
However, I am not really sure what the best way is. Maybe we should spend some time on https://dasch.myjetbrains.com/youtrack/issue/DSP-98

mdelez replied:
I'm currently writing unit tests for the PR mentioned above and the imports/injections (in test host components) for the FormBuilder are quite confusing. I think we should spend some time on what exactly needs to be imported in the unit tests that you don't get an injector error.

tobiasschweizer replied:
That is an excellent idea. Maybe we could try to work on https://dasch.myjetbrains.com/youtrack/issue/DSP-98 soon.
